omod/src/main/java/org/openmrs/web/controller/user/UserFormController.java

```diff
@@ -284,6 +291,19 @@ public String handleSubmission(WebRequest request, HttpSession httpSession, Mode
 				us.changeQuestionAnswer(user, secretQuestion, secretAnswer);
 			}
 			
+			//Check if admin wants the person associated with the user to be added to the Provider Table
+			if(addToProviderTableOption != null) {
+				Set<Role> userRoles = user.getRoles();
+				for (Role role : userRoles) {
+					if (role.getRole().equals("Provider")) { //If 'Provider' role is selected, we add the User to the Provider Table
```

Comment:
@wluyima I think it would make sense to add some JavaScript that automatically checks provider when this option is selected and unchecks it (if it was not checked previously) when unchecked